### DISCUSSION TITLE
How to make spatie/permissions checks in livewire component?

### DISCUSSION THREAD

**Discussion by PetroGromovo** [livewire/livewire#4148] [1. Help]
In Laravel 8 / livewire 2.5 I use       spatie/laravel-permission 5 and for CRUD operations I make checks in components like 

    public function edit()
    {
        $loggedUser = Auth::user();
    
        if ( $loggedUser->can(PERMISSION_EDIT_HOSTEL) ) {
            return view('livewire.error-msg', [
        ...

and I need to make such checks in any method of my component.
Looking at Lifecycle Hooks I wonder if there is a way to keep all checks in one place ?
How can I implement it ?

Thanks in advance!

**Comment by joshhanley** [comment]
@PetroGromovo you shouldn't return a view from action methods, as it won't work properly. Instead you should return a redirect or throw an error that you could display on screen to the user in a notification or something.

If you want to perform the same check on every request you could put it in the `booted` lifecycle method, see here https://laravel-livewire.com/docs/2.x/lifecycle-hooks

But you would need need to make sure your component only does one thing (like editing).

Hope this helps!

**Reply by PetroGromovo** [reply]
Thanks!

Debugging booted method, I see that is has 1 parameter like :

```
Livewire\Request Object
(
    [fingerprint] => Array
        (
            [id] => N88CNnpmNSd1xzBTuGQc
            [name] => admin.crud-items
            [locale] => en
            [path] => admin/items
            [method] => GET
        )

    [updates] => Array
        (
        )

    [memo] => Array
        (
        )

```

when I show listing(render method) and when I call "edit" method :

```
 Livewire\Request Object
        (
            [fingerprint] => Array
                (
                    [id] => N88CNnpmNSd1xzBTuGQc
                    [name] => admin.crud-items
                    [locale] => en
                    [path] => admin/items
                    [method] => GET
                )

            [updates] => Array
                (
                    [0] => Array
                        (
                            [type] => callMethod
                            [payload] => Array
                                (
                                    [id] => ki8a
                                    [method] => edit
                                    [params] => Array
                                        (
                                            [0] => 44
                                        )

                                )

                        )

                )
...

```

It permits me to make checks I need, but I doubt to use  booted method parameter, which is not described in docs.
Or there is some official docs on it ? How do you think is it good way/safe to use this way?
